hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/SqlToQueryType.java

```diff
@@ -69,6 +72,9 @@
 
         HZ_TO_CALCITE.put(QueryDataTypeFamily.OBJECT, SqlTypeName.ANY);
```

Comment:
If we agree to expose objects as `ANY`, then this should be reflected in the `01-type-system.md`

Reply:
After all the fixes applied in this PR there should be no mentions of `ANY` visible to users.